spec/support/capybara.rb

```diff
@@ -1,8 +1,11 @@
-require "capybara/poltergeist"
+require "selenium/webdriver"
 
-Capybara.register_driver :poltergeist do |app|
-  Capybara::Poltergeist::Driver.new(app, js_errors: true, inspector: true)
+Capybara.register_driver :headless_chrome do |app|
+  options = Selenium::WebDriver::Chrome::Options.new
+  options.add_argument("--headless")
+  options.add_argument("--window-size=1280,1024")
+  Capybara::Selenium::Driver.new(app, browser: :chrome, options: options)
```

Comment:
Some background and suggestions on options to pass here:

  - The `disable-gpu` flag is needed, at least as of a few months ago, based on [this article](https://developers.google.com/web/updates/2017/04/headless-chrome).

  - The `no-sandbox` is commonly added to help prevent crashing; see [this article](https://sites.google.com/a/chromium.org/chromedriver/help/chrome-doesn-t-start).

  - A large window-size turns out to be useful (and pretty much necessary) because Chrome is much more picky about what elements count as "visible" -- Chrome will throw an error claiming the element is not visible if it's outside of the viewport, whereas PhantomJS will implicitly scroll to it. So the solutions are to either fill up the test code with lots of calls to scroll the viewport to a certain place -- or simply use a large viewport to begin with.

Separately, I also have `logging_prefs` added to the driver config; that makes it possible to access JS logs from test code, which helps a lot with debugging.

The full capabilities object ends up looking like this:

```ruby
chrome_options = { "args" => ["headless", "disable-gpu", "window-size=2160,3840"] }

logging_prefs = { "browser": "ALL" }

capabilities = Selenium::WebDriver::Remote::Capabilities.chrome(
  chrome_options: options,
  logging_prefs: logging_prefs)
```

With the logging-prefs in place, the JS console output can be accessed from any test code through the `page` object. For example, this would print all of the console entries, one per line:

```ruby
page.driver.browser.manage.logs.get(:browser).map(&:inspect).join("\n")
```

Reply:
Thanks, will fold some of those in.

Found this regarding `--no-gpu` https://bugs.chromium.org/p/chromium/issues/detail?id=737678. TLDR; No longer needed.

Reply:
@paulrex SystemTests will be another PR. I'm not quite ready to do that and don't want to hold this one up.